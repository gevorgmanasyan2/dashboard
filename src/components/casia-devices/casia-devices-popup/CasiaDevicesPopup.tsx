import React, { FC } from 'react'
import Popup from '../../Popup/Popup'
import './casiaDevicesPopup.scss'
import { Link } from 'react-router-dom'

interface CasiaDevicesProps {
  className: string
  visible: boolean
  okText: string
  cancelText: string
  closeModal: () => void
}

const CasiaDevices: FC<CasiaDevicesProps> = ({ visible, closeModal }) => {
  return (
    <div className="casiadevices__container">
      <Popup
        className="casiadevices__container__modal"
        visible={visible}
        closeModal={closeModal}
        modalTitle="Your New Casia device"
        okButtonText={
          <Link to="/casiadevice" onClick={closeModal}>
            Let&apos;s Go!
          </Link>
        }
        cancelButtonText="Come back later"
      >
        <p className="casiadevices__container__modal--info">
          Our records show that you have a Casia Device that has not yet been configured
        </p>
        <p className="casiadevices__container__modal--offer">
          Would you like to get started now?(takes about 1 hour)
        </p>
      </Popup>
    </div>
  )
}

export default CasiaDevices
