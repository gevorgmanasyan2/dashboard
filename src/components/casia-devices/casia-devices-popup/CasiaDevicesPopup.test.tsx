/* eslint-disable no-undef */
import React from 'react'
import { shallow } from 'enzyme'
import CasiaDevicesPopup from './CasiaDevicesPopup'

it('should render CasiaDevicesPopup component', () => {
  const mockItem = {
    visible: true || false,
    className: 'casiadevices__container__modal',
    okText: 'Ok',
    cancelText: 'Cancel',
    // eslint-disable-next-line @typescript-eslint/no-empty-function
    closeModal: () => {},
  }
  expect(shallow(<CasiaDevicesPopup {...mockItem} />)).toMatchSnapshot()
})
