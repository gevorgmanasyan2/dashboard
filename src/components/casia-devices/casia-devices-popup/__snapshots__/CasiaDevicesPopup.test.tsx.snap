// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render CasiaDevicesPopup component 1`] = `
<div
  className="casiadevices__container"
>
  <Popup
    cancelButtonText="Come back later"
    className="casiadevices__container__modal"
    closeModal={[Function]}
    modalTitle="Your New Casia device"
    okButtonText={
      <Link
        onClick={[Function]}
        to="/casiadevice"
      >
        Let's Go!
      </Link>
    }
    visible={true}
  >
    <p
      className="casiadevices__container__modal--info"
    >
      Our records show that you have a Casia Device that has not yet been configured
    </p>
    <p
      className="casiadevices__container__modal--offer"
    >
      Would you like to get started now?(takes about 1 hour)
    </p>
  </Popup>
</div>
`;
